save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/ProjectInfo.kt

```diff
@@ -98,15 +127,11 @@ fun projectInfo(
                             type = InputType.text
                             this.id = fieldId
                             value = text ?: ""
-                            disabled = if (fieldId == "name") {
-                                // temporary workaround for https://github.com/saveourtool/save-cloud/issues/589#issuecomment-1049674021
-                                true
-                            } else {
-                                props.isEditDisabled ?: true
-                            }
+                            // temporary workaround for https://github.com/saveourtool/save-cloud/issues/589#issuecomment-1049674021
+                            disabled = fieldId != "name" && props.onProjectUpdate != null && isEditDisabled
+                            readOnly = fieldId == "name" || props.onProjectUpdate == null
                             onChange = { event ->
-                                val tg = event.target
-                                setDraftProject(idToValueSetter[fieldId]!!(tg.value))
+                                setDraftProject(idToValueSetter[fieldId]!!(event.target.value))
```

Comment:
you can use `getValue` to avoid usage of `!!`

Reply:
Thanks, will use it

Reply:
Done